simulation/simple_sensor_simulator/src/sensor_simulation/occupancy_grid/grid_cell.cpp

```diff
@@ -0,0 +1,283 @@
+    LineSegment(right_down, right_up), LineSegment(right_up, left_up)};
+  return ret;
+}
+
+std::vector<geometry_msgs::msg::Point> getIntersection2D(const std::vector<LineSegment> & lines)
+{
+  std::vector<geometry_msgs::msg::Point> ret;
+  for (size_t i = 0; i < lines.size(); i++) {
+    for (size_t m = 0; m < lines.size(); m++) {
+      if (i < m) {
+        const auto intersection = lines[i].getIntersection2D(lines[m]);
+        if (intersection) {
+          ret.emplace_back(intersection.get());
+        }
+      }
+    }
+  }
+  return ret;
+}
+
+geometry_msgs::msg::Point GridCell::transformToWorld(const geometry_msgs::msg::Point & point) const
+{
+  auto mat = quaternion_operation::getRotationMatrix(origin.orientation);
+  Eigen::VectorXd p(3);
+  p(0) = point.x;
+  p(1) = point.y;
+  p(2) = point.z;
+  p = mat * p;
+  p(0) = p(0) + origin.position.x;
+  p(1) = p(1) + origin.position.y;
+  p(2) = p(2) + origin.position.z;
+  geometry_msgs::msg::Point ret;
+  ret.x = p(0);
+  ret.y = p(1);
+  ret.z = p(2);
+  return ret;
+}
+
+bool GridCell::isIntersect2D(const LineSegment & line) const
+{
+  for (const auto & outside : getLineSegments()) {
+    if (outside.isIntersect2D(line)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+bool GridCell::isIntersect2D(const std::vector<LineSegment> & line_segments) const
+{
+  for (const auto & line_segment : line_segments) {
+    if (isIntersect2D(line_segment)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+bool GridCell::contains(const geometry_msgs::msg::Point & p) const
+{
+  const auto line_segments = getLineSegments();
+  typedef boost::geometry::model::d2::point_xy<double> boost_point;
+  typedef boost::geometry::model::polygon<boost_point> boost_polygon;
+  boost_polygon poly;
+  for (const auto & line_segment : line_segments) {
+    boost::geometry::exterior_ring(poly).push_back(
+      boost_point(line_segment.start_point.x, line_segment.start_point.y));
+  }
+  return boost::geometry::within(boost_point(p.x, p.y), poly);
+}
+
+bool GridCell::contains(const std::vector<geometry_msgs::msg::Point> & points) const
+{
+  for (const auto & point : points) {
+    if (contains(point)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+int8_t GridCell::getData() const { return data_; }
+void GridCell::setData(int8_t data) { data_ = data; }
+
+bool GridCell::operator==(const GridCell & rhs) const
+{
+  if (rhs.row == row && rhs.col == col && rhs.index == index) {
+    return true;
+  }
+  return false;
+}
+
+bool GridCell::operator!=(const GridCell & rhs) const { return !(*this == rhs); }
+
+bool GridCell::operator<(const GridCell & rhs) const { return index < rhs.index; }
+bool GridCell::operator>(const GridCell & rhs) const { return rhs < *this; }
+bool GridCell::operator<=(const GridCell & rhs) const { return !(rhs > *this); }
+bool GridCell::operator>=(const GridCell & rhs) const { return !(rhs < *this); }
+GridCell & GridCell::operator=(const GridCell &) { return *this; }
```

Comment:
Is this definition intentional?
If you want to invalidate the assignment operator, define it with `= delete`.